sign_csr.py

```diff
@@ -2,7 +2,43 @@
 import argparse, subprocess, json, os, urllib2, sys, base64, binascii, time, \
     hashlib, tempfile, re, copy, textwrap
 
-def sign_csr(pubkey, csr, email=None):
+
+def host_token(domain, token, response_payload, user_step_number, file_based):
```

Comment:
I don't agree that one huge function is more readable, but I've inlined it at your request.